pkg/flow/json_walker.go

```diff
@@ -0,0 +1,238 @@
+)
+
+// Field should probably spit this into block and non block fields
+type Field struct {
+	ID           string      `json:"id,omitempty"`
+	Key          string      `json:"key,omitempty"`
+	Name         string      `json:"name,omitempty"`
+	Label        string      `json:"label,omitempty"`
+	Type         string      `json:"type,omitempty"`
+	References   []string    `json:"references,omitempty"`
+	ReferencedBy []string    `json:"reference_by,omitempty"`
+	Health       *Health     `json:"health,omitempty"`
+	Original     string      `json:"original,omitempty"`
+	Value        interface{} `json:"value,omitempty"`
+}
+
+// ConvertBlock converts a set of component information into a generic Field json representation.
+func ConvertBlock(
+	id []string,
+	args component.Arguments,
+	exports component.Exports,
+	references, referencedby []string,
+	health *Health,
+	original string,
+) *Field {
+
+	nf := &Field{}
+	nf.Health = health
+	nf.References = references
+	nf.ReferencedBy = referencedby
+	nf.Original = original
+	nf.Type = "block"
+	nf.ID = strings.Join(id, ".")
+	nf.Name = strings.Join(id[0:2], ".")
+	nf.Label = id[2]
+	fields := make([]*Field, 0)
+	cArgs := convertArguments(args)
+	if args != nil {
+		fields = append(fields, cArgs)
+	}
+	cExports := convertExports(exports)
+	if exports != nil {
+		fields = append(fields, cExports)
+	}
+	nf.Value = fields
+	return nf
+}
+
+func convertArguments(args component.Arguments) *Field {
+	return convertField(args, &rivertags.Field{
+		Name:  []string{"arguments"},
+		Index: nil,
+		Flags: 0,
+	})
+}
+
+func convertExports(exports component.Exports) *Field {
+	return convertField(exports, &rivertags.Field{
+		Name:  []string{"exports"},
+		Index: nil,
+		Flags: 0,
+	})
+}
+
+// convertToField converts to a generic field for json
+func convertToField(in interface{}, name string) *Field {
+	return convertField(in, &rivertags.Field{
+		Name:  []string{name},
+		Index: nil,
+		Flags: 0,
+	})
+}
+
+func convertField(in interface{}, f *rivertags.Field) *Field {
+	// Assume everything is an attr unless otherwise specified
+	nf := &Field{
+		Type: "attr",
+	}
+	if f != nil && len(f.Name) > 0 {
+		nf.Key = f.Name[len(f.Name)-1]
+	}
+
+	nt := reflect.TypeOf(in)
+	vIn := reflect.ValueOf(in)
+	if in != nil {
+		for nt.Kind() == reflect.Pointer && !vIn.IsZero() {
+			vIn = vIn.Elem()
+			nt = vIn.Type()
+		}
+		in = vIn.Interface()
+	} else {
+		nf.Value = &Field{
+			Type: "null",
+		}
+		return nf
+	}
+
+	// Dont write zero value records
+	if reflect.ValueOf(in).IsZero() {
+		return nil
+	}
+
+	switch in.(type) {
+	case rivertypes.Secret:
+		nf.Value = &Field{
+			Type:  "string",
+			Value: "(secret)",
+		}
+		return nf
+	case rivertypes.OptionalSecret:
+		nf.Value = &Field{
+			Type:  "string",
+			Value: "(secret)",
+		}
+		maybeSecret := in.(rivertypes.OptionalSecret)
+		if !maybeSecret.IsSecret {
+			nf.Value.(*Field).Value = maybeSecret.Value
+		}
+		return nf
+	}
```

Comment:
This should be more generic; check to see if it calls RiverTokenize instead and tokenize it into a literal string to report as a capsule.